py3-cherrypy.yaml

```diff
@@ -0,0 +1,138 @@
+package:
+  name: py3-cherrypy
+  version: "18.10.0"
+  epoch: 0
+  description: "A pythonic, object-oriented HTTP framework,"
+  copyright:
+    - license: BSD-3-clause
+  dependencies:
+    provider-priority: 0
+
+vars:
+  pypi-package: cherrypy
+
+data:
+  - name: py-versions
+    items:
+      3.10: "310"
+      3.11: "311"
+      3.12: "312"
+      3.13: "313"
+
+environment:
+  contents:
+    packages:
+      - py3-supported-build-base-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/cherrypy/cherrypy
+      tag: v${{package.version}}
+      expected-commit: d1eb64e27ac2dbb761b3cf96eb0c883d5bd87786
+
+  - uses: patch
+    with:
+      patches: filelock.patch
+
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provides:
+        - py3-${{vars.pypi-package}}
+      provider-priority: ${{range.value}}
+      runtime:
+        - py${{range.key}}-cheroot
+        - py${{range.key}}-portend
+        - py${{range.key}}-filelock
+        - py${{range.key}}-jaraco.collections
+        - py${{range.key}}-more-itertools
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+      - name: move usr/bin executables for -bin
+        runs: |
+          mkdir -p ./.cleanup/${{range.key}}/
+          mv ${{targets.contextdir}}/usr/bin ./.cleanup/${{range.key}}/
+      - uses: strip
+    test:
+      environment:
+        contents:
+          packages:
+            - py${{range.key}}-pytest
+            - py${{range.key}}-path
+            - py${{range.key}}-requests
+            - py${{range.key}}-requests-toolbelt
+      pipeline:
+        - uses: python/import
+          with:
+            python: python${{range.key}}
+            import: ${{vars.pypi-package}}
+        - uses: git-checkout
+          with:
+            repository: https://github.com/cherrypy/cherrypy
+            tag: v${{package.version}}
+            expected-commit: d1eb64e27ac2dbb761b3cf96eb0c883d5bd87786
+        - uses: patch
```

Comment:
Any reason why you created a hand-made patch instead of using `git-checkout`'s 'cherry-picks:`? It's a cleaner alternative, if the patch is part of the upstream repo. Might be the case here, right?

Reply:
It is - I discovered this feature this morning! I'll re-do with cherry-picks instead plus the pytest-xdist package should be available shortly to reduce test durations